File: BackEnd_Learn/Java/sms/src/main/java/com/lee/sms/error/exception/ErrorCode.java
package com.lee.sms.error.exception;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
@JsonFormat(shape = JsonFormat.Shape.OBJECT)
public enum ErrorCode {

    NUMBER_NOT_EXIST(404, "verify number not exist."),
    EXPIRED_AUTH_CODE(400, "verify was expired."),
    USER_SAVE(500, "User Save Error."),
    NUMBER_NOT_MATCH(400, "verify number not match");

    private final int status;
    private final String message;
}

File: BackEnd_Learn/kotlin/auth/src/main/kotlin/io/github/leeseojune53/auth/error/exception/ErrorCode.kt
package io.github.leeseojune53.auth.error.exception

import com.fasterxml.jackson.annotation.JsonFormat

@JsonFormat(shape = JsonFormat.Shape.STRING)
enum class ErrorCode(
    private val statusCode: Int,
    private val message: String
) {
    USER_NOT_FOUND(404, "User not found.");

    fun getStatusCode(): Int {
        return statusCode
    }

    fun getMessage(): String {
        return message
    }

}